lib/src/utils.ts

```diff
@@ -139,3 +140,32 @@ export async function extractPemFromKeyString(keyString: string): Promise<string
 
   return pem;
 }
+
+/**
+ * Extracts the error message from an RPC catch error.
+ */
+export function extractRpcErrorMessage(error: unknown): string {
+  if (error instanceof ConnectError || error instanceof Error) {
+    return error.message;
+  }
+  return 'Unknown network error occurred';
+}
+
+/**
+ * Converts a KAS endpoint URL to a platform URL.
+ * If the KAS endpoint ends with '/kas', it returns the host url
+ * Otherwise, it returns the original KAS endpoint.
+ */
+export function getPlatformUrlFromKasEndpoint(endpoint: string): string {
+  let result = endpoint || '';
+  if (result.endsWith('/')) {
+    result = rstrip(result, '/');
+  }
+  if (result.endsWith('/kas')) {
+    result = result.slice(0, -4);
+  }
+  if (result.endsWith('/v2/rewrap')) {
+    result = result.slice(0, -10);
+  }
```

Comment:
Shouldn't these be flipped?
```suggestion
  if (result.endsWith('/v2/rewrap')) {
    result = result.slice(0, -10);
  }
  if (result.endsWith('/kas')) {
    result = result.slice(0, -4);
  }
```

Reply:
Thats a good catch, within tests it hasn't been a problem but I can see how if the url is `/kas/v2/rewrap` we do want this flipped